### ISSUE TITLE
Support nextval in distributed queries

### ISSUE THREAD

**Issue by marcocitus** [citusdata/citus#2304]
Many users run into the lack of `nextval` support in distributed queries, especially when using INSERT...SELECT into a table which has a `serial` column, in which case Postgres automatically adds the call to `nextval` to the SELECT query. If the SELECT query is distributed, the call to `nextval` will fail on the worker because the sequence does not exist.

```sql
SELECT nextval('test_x_seq'::regclass) FROM test;
WARNING:  relation "public.test_x_seq" does not exist
ERROR:  failed to execute task 3
```

One option that we can already use for `bigserial` columns in MX is to create the sequence on each node, but set it to a different integer range. That way, the sequence will still return unique values, though they will no longer follow a consistent order. It will also cause some complications when relying on [sequence values for rollups](https://www.citusdata.com/blog/2018/06/14/scalable-incremental-data-aggregation/), though this is solvable through some more extra bookkeeping. Due to the extra bits, this approach only works for `bigserial`, not smaller `serial` columns.

Another alternative is to always evaluate the `nextval` on the coordinator by evaluating it in the master plan. This is much slower, but behaves more like postgres. 

For a distributed INSERT...SELECT with a call to nextval, a possible optimisation is to first write the query result into a set of intermediate results on the worker, generate the required number of sequence values on the coordinator for each intermediate result, and then write the intermediate results along with the sequence values into the destination table.

**Comment by onderkalaci** [comment]
Can we close this with #2572 @marcocitus?

**Comment by marcocitus** [comment]
No, the function call might not be in a subquery.

**Comment by saicitus** [comment]
Customer has run into a similar issue with some differences: 

INSERT INTO SELECT was between the same table (source and destination was same) but the distribution column was being changed in the INSERT INTO SELECT. The goal for the customer was to de-duplicate data with a different sharding column and obviously increasing id sequence.

```
create table a_seq(id serial, dist_key int);
SELECT create_distributed_table('a_seq','dist_key');
INSERT INTO a_seq(dist_key) values (1);
INSERT INTO a_seq(dist_key) select dist_key*100 from a_seq;
ERROR:  INSERT ... SELECT cannot generate sequence values when selecting from a distributed table
```

**Comment by marcocitus** [comment]
Related to#2365

**Comment by marcocitus** [comment]
Fixed by #3680
